AppLovinMAXDemo/Plugins/AppLovinMAX/Source/AppLovinMAX/Public/AppLovinMAXDelegate.h

```diff
@@ -0,0 +1,152 @@
+// Copyright AppLovin Corporation. All Rights Reserved.
+
+#pragma once
+
+#include "CoreMinimal.h"
+#include "AdError.h"
+#include "AdInfo.h"
+#include "AdReward.h"
+#include "SdkConfiguration.h"
+#include "AppLovinMAXDelegate.generated.h"
+
+DECLARE_DYNAMIC_MULTICAST_DELEGATE_OneParam(FOnSdkInitializedDynamicDelegate, const FSdkConfiguration &, SdkConfiguration);
+
+DECLARE_DYNAMIC_MULTICAST_DELEGATE_OneParam(FOnBannerAdLoadedDynamicDelegate, const FAdInfo &, AdInfo);
+DECLARE_DYNAMIC_MULTICAST_DELEGATE_TwoParams(FOnBannerAdLoadFailedDynamicDelegate, const FAdInfo &, AdInfo, const FAdError &, AdError);
+DECLARE_DYNAMIC_MULTICAST_DELEGATE_OneParam(FOnBannerAdClickedDynamicDelegate, const FAdInfo &, AdInfo);
+DECLARE_DYNAMIC_MULTICAST_DELEGATE_OneParam(FOnBannerAdExpandedDynamicDelegate, const FAdInfo &, AdInfo);
+DECLARE_DYNAMIC_MULTICAST_DELEGATE_OneParam(FOnBannerAdCollapsedDynamicDelegate, const FAdInfo &, AdInfo);
+DECLARE_DYNAMIC_MULTICAST_DELEGATE_OneParam(FOnBannerAdRevenuePaidDynamicDelegate, const FAdInfo &, AdInfo);
+
+DECLARE_DYNAMIC_MULTICAST_DELEGATE_OneParam(FOnMRecAdLoadedDynamicDelegate, const FAdInfo &, AdInfo);
+DECLARE_DYNAMIC_MULTICAST_DELEGATE_TwoParams(FOnMRecAdLoadFailedDynamicDelegate, const FAdInfo &, AdInfo, const FAdError &, AdError);
+DECLARE_DYNAMIC_MULTICAST_DELEGATE_OneParam(FOnMRecAdClickedDynamicDelegate, const FAdInfo &, AdInfo);
+DECLARE_DYNAMIC_MULTICAST_DELEGATE_OneParam(FOnMRecAdExpandedDynamicDelegate, const FAdInfo &, AdInfo);
+DECLARE_DYNAMIC_MULTICAST_DELEGATE_OneParam(FOnMRecAdCollapsedDynamicDelegate, const FAdInfo &, AdInfo);
+DECLARE_DYNAMIC_MULTICAST_DELEGATE_OneParam(FOnMRecAdRevenuePaidDynamicDelegate, const FAdInfo &, AdInfo);
+
+DECLARE_DYNAMIC_MULTICAST_DELEGATE_OneParam(FOnInterstitialLoadedDynamicDelegate, const FAdInfo &, AdInfo);
+DECLARE_DYNAMIC_MULTICAST_DELEGATE_TwoParams(FOnInterstitialLoadFailedDynamicDelegate, const FAdInfo &, AdInfo, const FAdError &, AdError);
+DECLARE_DYNAMIC_MULTICAST_DELEGATE_OneParam(FOnInterstitialHiddenDynamicDelegate, const FAdInfo &, AdInfo);
```

Comment:
nit but I would've kept the ordering of callbacks consistent with other fullscreen formats, and the delegate declarations below.